src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -589,6 +671,61 @@ public static int IndexOfAny<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> value
                     ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(values)),
                     values.Length);
 
+            if (typeof(T) == typeof(char))
+            {
+                ref var valueRef = ref Unsafe.As<T, char>(ref MemoryMarshal.GetReference(values));
+                if (values.Length == 5)
+                {
+                    // Length 5 is a common length for FileSystemName expression (", <, >, *, ?) and in preference to 2 as it has an explicit overload
```

Comment:
@JeremyKuhne helpful to have this first? `FileSystemName` switches between two different length arrays 5 and 2

Reply:
Sorry, missed this. I hope it becomes less common for 5 in the future, but for now it is by far and away the most common case. The new default when using `EnumerationOptions` is the simple type (i.e. 2).